### ISSUE TITLE
Docker-compose configuration for traefik reverse proxy and authentik identity provider over https

### ISSUE THREAD

**Issue by chrda81** [sartography/spiff-arena#449]
Hi,

I try to get the spiff-arena working on a server with your provided docker-compose.yml. When I open the URL http://<server_name>.<domain_name>.com:8001 the following URL redirection is shown in my browsers address bar: https://api.<server_name>.<domain_name>.com/v1.0/login?redirect_url=http%3A%2F%2F<server_name>.<domain_name>.com%3A8001%2F which cannot be resolved.

Do I have to create the DNS entry for api.<server_name>.<domain_name>.com, or can I rewrite the URL, so that the spiff-arena is working on that server?

It would be great if you could help me to configure the compose file for traefik reverse proxy and authentik as identity provider, to get the spiff-arena working on my domain with https. I created a traefik-crowdsec-stack with authentik based on this tutorial page https://goneuland.de/traefik-v2-3-reverse-proxy-mit-crowdsec-im-stack-einrichten/ and this stack is working really great with other docker services.

I'm grateful for every little hint

**Comment by danfunk** [comment]
You do not have to use the api.<server_name>, but that is the default behavior.  You can configure the backend url at the moment the docker container for the front end is deployed.  You will need to set SPIFFWORKFLOW_FRONTEND_RUNTIME_CONFIG_BACKEND_BASE_URL  in the spiffworkflow-frontend environment variables in your docker compose (right about [here](https://github.com/sartography/spiff-arena/blob/main/docker-compose.yml#L10))   You should see that the base URL the front end is trying to call is now the endpoint you specified.

As for Authentik, we have not tested against it, but we have tested against both Google OpenID and KeyCloak's OpenID.  It looks like Authentik does support OpenID, so you should be able to update those settings in the docker compose file as well, on lines 30, 31 and 32 of the same file as above.

**Comment by chrda81** [comment]
Many thanks Dan!

This is my working docker-compose.yml for the Traefik-CrowdSec-Stack redirecting to https:

```yml
version: "3.8"
services:
  spiffworkflow-frontend:
    container_name: spiffworkflow-frontend
    image: ghcr.io/sartography/spiffworkflow-frontend:latest
    depends_on:
      spiffworkflow-backend:
        condition: service_healthy
    environment:
      APPLICATION_ROOT: "/"
      PORT0: "${SPIFFWORKFLOW_FRONTEND_PORT:-8001}"
      SPIFFWORKFLOW_FRONTEND_RUNTIME_CONFIG_BACKEND_BASE_URL: "http://<server_name>.<domain_name>.com:${SPIFF_BACKEND_PORT:-8>
    ports:
      - "${SPIFFWORKFLOW_FRONTEND_PORT:-8001}:${SPIFFWORKFLOW_FRONTEND_PORT:-8001}/tcp"
    networks:
      - default
      - proxy
    labels:
      - "traefik.http.routers.spiffWeb-https.tls.certresolver=http_resolver"
      - "traefik.docker.network=proxy"
      - "traefik.constraint-label=proxy"
      - "traefik.enable=true"
      - "traefik.http.middlewares.redirect-http-to-https.redirectScheme.scheme=https"
      - "traefik.http.middlewares.redirect-http-to-https.redirectScheme.permanent=true"
      - "traefik.http.routers.spiffWeb-https.rule=Host(`<server_name>.<domain_name>.com`)"
      - "traefik.http.routers.spiffWeb-https.entrypoints=websecure"
      - "traefik.http.routers.spiffWeb-https.tls=true"
      - "traefik.http.routers.spiffWeb-https.service=spiffWeb"
      - "traefik.http.routers.spiffWeb-http.rule=Host(`<server_name>.<domain_name>.com`)"
      - "traefik.http.routers.spiffWeb-http.entrypoints=web"
      - "traefik.http.routers.spiffWeb-http.middlewares=redirect-http-to-https"
      - "traefik.http.routers.spiffWeb-http.service=spiffWeb"
      - "traefik.http.services.spiffWeb.loadbalancer.server.scheme=http"
      - "traefik.http.services.spiffWeb.loadbalancer.server.port=8001"

  spiffworkflow-backend:
    container_name: spiffworkflow-backend
    image: ghcr.io/sartography/spiffworkflow-backend:latest
    environment:
      SPIFFWORKFLOW_BACKEND_APPLICATION_ROOT: "/"
      SPIFFWORKFLOW_BACKEND_ENV: "local_development"
      FLASK_DEBUG: "0"
      FLASK_SESSION_SECRET_KEY: "${FLASK_SESSION_SECRET_KEY:-super_secret_key}"
      # WARNING: Frontend is a static site which assumes frontend port - 1 on localhost.
      SPIFFWORKFLOW_BACKEND_URL: "http://<server_name>.<domain_name>.com:${SPIFF_BACKEND_PORT:-8000}"
      SPIFFWORKFLOW_BACKEND_BPMN_SPEC_ABSOLUTE_DIR: "/app/process_models"
      SPIFFWORKFLOW_BACKEND_CONNECTOR_PROXY_URL: "http://spiffworkflow-connector:8004"
      SPIFFWORKFLOW_BACKEND_DATABASE_TYPE: "sqlite"
      SPIFFWORKFLOW_BACKEND_DATABASE_URI: "sqlite:////app/db_volume/db.sqlite3"
      SPIFFWORKFLOW_BACKEND_LOAD_FIXTURE_DATA: "false"
      SPIFFWORKFLOW_BACKEND_OPEN_ID_CLIENT_ID: "spiffworkflow-backend"
      SPIFFWORKFLOW_BACKEND_OPEN_ID_CLIENT_SECRET_KEY: "my_open_id_secret_key"
      SPIFFWORKFLOW_BACKEND_OPEN_ID_SERVER_URL: "http://<server_name>.<domain_name>.com:${SPIFF_BACKEND_PORT:-8000}/application/o/<application_name>/"
      SPIFFWORKFLOW_BACKEND_PERMISSIONS_FILE_NAME: "example.yml"
      SPIFFWORKFLOW_BACKEND_PORT: "${SPIFF_BACKEND_PORT:-8000}"
      SPIFFWORKFLOW_BACKEND_RUN_BACKGROUND_SCHEDULER: "true"
      SPIFFWORKFLOW_BACKEND_UPGRADE_DB: "true"
      SPIFFWORKFLOW_BACKEND_URL_FOR_FRONTEND: "http://<server_name>.<domain_name>.com:${SPIFFWORKFLOW_FRONTEND_PORT:-8001}"
    ports:
      - "${SPIFF_BACKEND_PORT:-8000}:${SPIFF_BACKEND_PORT:-8000}/tcp"
    volumes:
      - ./process_models:/app/process_models
      - spiffworkflow_backend_db:/app/db_volume
    healthcheck:
      test: "curl localhost:${SPIFF_BACKEND_PORT:-8000}/v1.0/status --fail"
      interval: 10s
      timeout: 5s
      retries: 20
    networks:
      - default

  spiffworkflow-connector:
    container_name: spiffworkflow-connector
    image: ghcr.io/sartography/connector-proxy-demo:latest
    environment:
      FLASK_ENV: "${FLASK_ENV:-development}"
      FLASK_DEBUG: "0"
      FLASK_SESSION_SECRET_KEY: "${FLASK_SESSION_SECRET_KEY:-super_secret_key}"
      CONNECTOR_PROXY_PORT: "${SPIFF_CONNECTOR_PORT:-8004}"
    ports:
      - "${SPIFF_CONNECTOR_PORT:-8004}:${SPIFF_CONNECTOR_PORT:-8004}/tcp"
    healthcheck:
      test: "curl localhost:${SPIFF_CONNECTOR_PORT:-8004}/liveness --fail"
      interval: 10s
      timeout: 5s
      retries: 20
    networks:
      - default

volumes:
  spiffworkflow_backend_db:
    driver: local

networks:
  proxy:
    external: true
```

My Authentik OpenID provider is working as well. The correct URL is shown in the Authentik admin portal, when configuring the provider.

**Comment by chrda81** [comment]
One more question for the identity provider config. It is working as mentioned above. But how can I give a user from OpenID  admin privileges?

**Comment by chrda81** [comment]
Thanks for the link. What I did:

1.) Extending the docker-compose.yml

```yml
spiffworkflow-backend:
  ..
  environment:
    SPIFFWORKFLOW_BACKEND_PERMISSIONS_FILE_NAME: "example.yml"
    ..
  volumes:
    - ./permissions:/app/src/spiffworkflow_backend/config/permissions
```

2.) Modified the file ./permissions/example.yml according to your documentation

Result: The permission file is loaded but it seems that admin rights are still missing -> The button "Add process group" is missing, as are the other menu entries like "Configuration": 
![image](https://github.com/sartography/spiff-arena/assets/36750848/3d15df82-5e7e-478d-aba0-26b55c7ca71b)


So I started with the docker-compose.yml over again and use the settings for your local openid provider according to https://raw.githubusercontent.com/sartography/spiff-arena/main/docker-compose.yml. I cleared my dockers db volume and started everything and after login the admin user has also no admin rights. I started the configuration in my local docker host and everything is working. So what is the root cause for that different behaviour between the local instance and the one on my server (serving with ssl)? Please find some logfiles attached.

[_spiffworkflow-backend_logs.txt](https://github.com/sartography/spiff-arena/files/12436555/_spiffworkflow-backend_logs.txt)
[_spiffworkflow-frontend_logs.txt](https://github.com/sartography/spiff-arena/files/12436557/_spiffworkflow-frontend_logs.txt)
[_spiffworkflow-connector_logs.txt](https://github.com/sartography/spiff-arena/files/12436558/_spiffworkflow-connector_logs.txt)

**Comment by burnettk** [comment]
there is definitely some unhappiness in the backend log file you linked. when the buttons and nav elements are missing, is backend up? like if you add `/v1.0/status` to the backend path, do you get some json? it is quite challenging to debug async like this, but yeah, if you have it running locally, it is certainly possible to get it running on a server. feel free to reach out on discord (https://discord.com/invite/F6Kb7HNK7B) if you would like to try to live debug. we're looking for customers who can help sustain the product, so we can try to be helpful. :)

**Comment by chrda81** [comment]
Many thanks Kevin, you spotted in a right direction. I put in the environment variable SPIFFWORKFLOW_FRONTEND_RUNTIME_CONFIG_BACKEND_BASE_URL as Dan mentioned to get rid of the api-call error. But then the frontend wasn't able to get a connection to the backend. What I did is to set up another subdomain for the backend and configure the traefik labels for it to route the port 443 to 8000, because I wanted to use SSL.

Here is the modified docker-compose.yml:
```yml
version: "3.8"
services:
  spiffworkflow-frontend:
    environment:
    ..
    SPIFFWORKFLOW_FRONTEND_RUNTIME_CONFIG_BACKEND_BASE_URL: "https://workflow.<domain_name>.com"
    ..
    networks:
      - default
      - proxy
    labels:
      traefik.docker.network: proxy
      traefik.enable: true
      traefik.http.routers.spiffWeb-secure.entrypoints: websecure
      traefik.http.routers.spiffWeb-secure.rule: Host(`workflow.<domain_name>.com`)
      traefik.http.routers.spiffWeb-secure.service: spiffWeb
      traefik.http.routers.spiffWeb-secure.tls: true
      traefik.http.routers.spiffWeb-secure.tls.certresolver: http_resolver
      traefik.http.routers.spiffWeb.entrypoints: web
      traefik.http.routers.spiffWeb.rule: Host(`workflow.<domain_name>.com`)
      traefik.http.services.spiffWeb.loadbalancer.server.port: 8001
      traefik.http.routers.spiffWeb.middlewares: default@file

  spiffworkflow-backend:
    ..
    environment:
      SPIFFWORKFLOW_BACKEND_URL: "https://workflow-backend.<domain_name>.com"
      SPIFFWORKFLOW_BACKEND_URL_FOR_FRONTEND: "https://workflow.<domain_name>.com"
      ..
    networks:
      - default
      - proxy
    labels:
      traefik.docker.network: proxy
      traefik.enable: true
      traefik.http.routers.spiffBackend-secure.entrypoints: websecure
      traefik.http.routers.spiffBackend-secure.rule: Host(`workflow-backend.<domain_name>.com`)
      traefik.http.routers.spiffBackend-secure.service: spiffBackend
      traefik.http.routers.spiffBackend-secure.tls: true
      traefik.http.routers.spiffBackend-secure.tls.certresolver: http_resolver
      traefik.http.routers.spiffBackend.entrypoints: web
      traefik.http.routers.spiffBackend.rule: Host(`workflow-backend.<domain_name>.com`)
      traefik.http.services.spiffBackend.loadbalancer.server.port: 8000
      traefik.http.routers.spiffBackend.middlewares: default@file
```

So the authorization and permissions are working for a user which is put into my permissions file. But what is not working is a placeholder for "everybody" who is not defined in the permissions file and should be able to log into the portal seeing the basic stuff. I tried:

```yml
  basic-permissions:
    groups: [ everybody ]
    allowed_permissions: [ all ]
    uri: BASIC
```
and

```yml
  basic-permissions:
    groups: [ everybody ]
    allowed_permissions: [ all ]
    uri: /login*
```

but nothing is working. The user I tried is not listet in the users section of the permission file, so the site url is bouncing between trying this user to login. Do you have a hint for that?

**Comment by burnettk** [comment]
ok, perhaps the issue is that the "everybody" group does not include everybody. since you mentioned you have modified the permissions file, you can assign users to groups like on line 25 in this file: https://github.com/sartography/spiff-arena/blob/main/spiffworkflow-backend/src/spiffworkflow_backend/config/permissions/example.yml

if this is not the issue, please confirm that the usernames in the db (`user` table. and see also `select * from user_group_assignment`) are matching the entries in the example.yml permissions file.

**Comment by chrda81** [comment]
TL;DR:

How do I define the mapping for a * users relation to group 'everybody', so that every user defined in my identity provider which logs into the frontend has basic view permissions?

The long story:
Ok, let me explain my thoughts. I have some users and groups defined in my identity provider. I want to rely on this configuration and manage which user is able to log into the SpiffWorkflow frontend. I have a identity provider group for 'IT Admins' and 'DevOps'. Every user in the 'IT Admin' group should become a admin when they log into and all of group 'DevOps' should become the permissions to develop and run workflows in their process group, but don't have the elevated permissions to modify things in other process groups. All other users defined in the identity provider which can start the SpiffWorkflow frontend should be able to login with basic view permissions.

I try to find the correct permission mapping for that scenario. I found the example workflow [set-permissions](https://github.com/sartography/sample-process-models/tree/sample-models-1/site-administration/set-permissions) where r".*" is declared as placeholder for the group 'everybody'. So my permission file only contains the following lines:

```yml
groups:
  admin:
    users:
      - admin@<domain_name>.com

permissions:
  # Admins have access to everything.
  admin:
    groups: [ admin ]
    allowed_permissions: [ create, read, update, delete ]
    uri: /*

  basic-permissions:
    groups: [ everybody ]
    allowed_permissions: [ all ]
    uri: /login*
```

I thought the r".*" I declared with the set-permissions workflow is taking care of the rest? But I got a login loop for all other users than 'admin@<domain_name>.com'.

**Comment by jasquat** [comment]
We do not actually support `r".*"` for assigning users to groups. Instead there is a config set in the backend that defines a default group that everyone goes into - "SPIFFWORKFLOW_BACKEND_DEFAULT_USER_GROUP". It defaults to "everybody". We will remove that regex line from our sample models to avoid confusion.

In order to give basic permissions to this everybody group, what you have is fine except change the uri from "/login*" to "BASIC" and a macro on the backend will populate the permissions when the server boots up. These macros are explained at https://spiff-arena.readthedocs.io/en/latest/DevOps_installation_integration/permission_url.html.

**Comment by chrda81** [comment]
Thanks jasquat, I did update my config with your changes and in the database tables for users, groups and permissions everything looks like expected. But when I try to login via my OpenID provider (Authentik.io) I still got some weird behaviour. A notification popup from my OpenID backend says, that the user credentials are correct and then the redirect to the spiff frontend is triggerd. But then the spiff backend returns its uri and it seems that it is thrown back to the OpenID login, which redirects to the spiff frontend and every is caught up in a loop. Only one user is working, which was set up as admin. I tried the other user with the weird behaviour to set up as admin as well, but the login loop is starting again. I don't know exactly how to find out, which causes this, but I'm using the same OpenID provider for three other web applications and there it is working. Do you have any hints for me, for what phrases I should look in the logfile?

https://github.com/sartography/spiff-arena/assets/36750848/48781ba2-d3fe-43c6-9f8b-2ad86c272d27

**Comment by burnettk** [comment]
so that TOKEN INVALID thing means one of four things went wrong. i added some logging that will hopefully point to the specific thing. if you can run the docker compose pull and start up fresh again, you will get this new logging: https://github.com/sartography/spiff-arena/commit/a4364c30a6110b87101dc4a87e75461ac81e4669

**Comment by chrda81** [comment]
Thank you so much, Kevin! Now this is working for my localhost setup. The new logging said: "TOKEN INVALID because ISS 'http://host.docker.internal:9000/application/o/spiff-arena/' does not match server url 'http://host.docker.internal:9000/application/o/spiff-arena'" .. One missing / at the end 😅

But I have to do some more digging on my server, where I'm trying to serve everything with SSL and a reverse proxy. There is currently no logging for this case. This means the login loop is happening earlier. I'm still investigating...

**Comment by chrda81** [comment]
Ok, I'm totally confused. The local docker setup with my ID Provider is working. But when I try to use this setup on a remote server (it doesn't matter if http or https) I'm redirected to the login page - the redirection loop is started. So I now tried to use the local open id backend for testing, but I'm redirected to the login page as well. Do you have a working docker-compose setup for running the Spiff-Arena on a remote server?

Here is my used docker-compose.yml

```yml
version: "3.8"
services:
  spiffworkflow-db:
    container_name: spiffworkflow-db
    image: mariadb:10.6
    platform: linux/amd64
    cap_add:
      - SYS_NICE
    restart: unless-stopped
    environment:
      - MYSQL_DATABASE=${SPIFFWORKFLOW_BACKEND_DATABASE_NAME:-spiffworkflow_backend_development}
      - MYSQL_ROOT_PASSWORD=${SPIFFWORKFLOW_BACKEND_MYSQL_ROOT_PASSWORD:-my-secret-pw}
      - MYSQL_TCP_PORT=7003
    ports:
      - "${ZITI_INTERFACE:-0.0.0.0}:7003:7003"
    volumes:
      - spiffworkflow_backend_db:/var/lib/mysql
    healthcheck:
      test: mysql --user=root --password='${SPIFFWORKFLOW_BACKEND_MYSQL_ROOT_PASSWORD:-my-secret-pw}' -e 'select 1' ${SPIFFWORKFLOW_BACKEND_DATABASE_NAME:-spiffworkflow_backend_development}
      interval: 10s
      timeout: 5s
      retries: 10
    networks:
      - default

  spiffworkflow-frontend:
    container_name: spiffworkflow-frontend
    image: ghcr.io/sartography/spiffworkflow-frontend:latest
    depends_on:
      spiffworkflow-backend:
        condition: service_healthy
    environment:
      APPLICATION_ROOT: "/"
      PORT0: "${SPIFFWORKFLOW_FRONTEND_PORT:-8001}"
      SPIFFWORKFLOW_FRONTEND_RUNTIME_CONFIG_BACKEND_BASE_URL: "http://spiffarena-backend.<domain_name>.ziti:${SPIFF_BACKEND_PORT:-8000}"
    #ports:
    #  - "${SPIFFWORKFLOW_FRONTEND_PORT:-8001}:${SPIFFWORKFLOW_FRONTEND_PORT:-8001}/tcp"
    networks:
      zitiblue:
        aliases:
          - spiffworkflow-frontend.blue
          - spiffarena.<domain_name>.ziti

  spiffworkflow-backend:
    container_name: spiffworkflow-backend
    image: ghcr.io/sartography/spiffworkflow-backend:latest
    depends_on:
      spiffworkflow-db:
        condition: service_healthy
    environment:
      SPIFFWORKFLOW_BACKEND_APPLICATION_ROOT: "/"
      SPIFFWORKFLOW_BACKEND_ENV: "local_docker"
      FLASK_DEBUG: "0"
      FLASK_SESSION_SECRET_KEY: "${FLASK_SESSION_SECRET_KEY:-super_secret_key}"
      # WARNING: Frontend is a static site which assumes frontend port - 1 on localhost.
      SPIFFWORKFLOW_BACKEND_URL: "http://spiffarena-backend.<domain_name>.ziti:${SPIFF_BACKEND_PORT:-8000}"
      SPIFFWORKFLOW_BACKEND_BPMN_SPEC_ABSOLUTE_DIR: "/app/process_models"
      SPIFFWORKFLOW_BACKEND_CONNECTOR_PROXY_URL: "http://spiffworkflow-connector:8004"
      SPIFFWORKFLOW_BACKEND_DATABASE_URI: "mysql+mysqldb://root:${SPIFFWORKFLOW_BACKEND_MYSQL_ROOT_PASSWORD:-my-secret-pw}@spiffworkflow-db:7003/${SPIFFWORKFLOW_BACKEND_DATABASE_NAME:-spiffworkflow_backend_development}"
      SPIFFWORKFLOW_BACKEND_LOAD_FIXTURE_DATA: false
      SPIFFWORKFLOW_BACKEND_LOG_LEVEL: "DEBUG"
      SPIFFWORKFLOW_BACKEND_OPEN_ID_CLIENT_ID: "spiffworkflow-backend"
      SPIFFWORKFLOW_BACKEND_OPEN_ID_CLIENT_SECRET_KEY: "my_open_id_secret_key"
      SPIFFWORKFLOW_BACKEND_OPEN_ID_SERVER_URL: "http://spiffarena-backend.<domain_name>.ziti:${SPIFF_BACKEND_PORT:-8000}/openid"
      SPIFFWORKFLOW_BACKEND_PERMISSIONS_FILE_NAME: "example.yml"
      SPIFFWORKFLOW_BACKEND_PORT: "${SPIFF_BACKEND_PORT:-8000}"
      SPIFFWORKFLOW_BACKEND_RUN_BACKGROUND_SCHEDULER_IN_CREATE_APP: false
      SPIFFWORKFLOW_BACKEND_UPGRADE_DB: true
      SPIFFWORKFLOW_BACKEND_URL_FOR_FRONTEND: "http://spiffarena.<domain_name>.ziti:${SPIFFWORKFLOW_FRONTEND_PORT:-8001}"
      SPIFFWORKFLOW_BACKEND_CHECK_FRONTEND_AND_BACKEND_URL_COMPATIBILITY: false
    #ports:
    #  - "${ZITI_INTERFACE:-0.0.0.0}:${SPIFF_BACKEND_PORT:-8000}:${SPIFF_BACKEND_PORT:-8000}/tcp"
    volumes:
      - ./process_models:/app/process_models
      # - ./permissions:/app/src/spiffworkflow_backend/config/permissions
      - spiffworkflow_backend_db:/var/lib/mysql
    healthcheck:
      test: "curl localhost:${SPIFF_BACKEND_PORT:-8000}/v1.0/status --fail"
      interval: 10s
      timeout: 5s
      retries: 20
    networks:
      default:
      zitiblue:
        aliases:
          - spiffworkflow-backend.blue
          - spiffarena-backend.<domain_name>.ziti

  spiffworkflow-connector:
    container_name: spiffworkflow-connector
    image: ghcr.io/sartography/connector-proxy-demo:latest
    environment:
      FLASK_ENV: "${FLASK_ENV:-development}"
      FLASK_DEBUG: "0"
      FLASK_SESSION_SECRET_KEY: "${FLASK_SESSION_SECRET_KEY:-super_secret_key}"
      CONNECTOR_PROXY_PORT: "${SPIFF_CONNECTOR_PORT:-8004}"
    ports:
      - "${ZITI_INTERFACE:-0.0.0.0}:${SPIFF_CONNECTOR_PORT:-8004}:${SPIFF_CONNECTOR_PORT:-8004}/tcp"
    healthcheck:
      test: "curl localhost:${SPIFF_CONNECTOR_PORT:-8004}/liveness --fail"
      interval: 10s
      timeout: 5s
      retries: 20
    networks:
      - default

volumes:
  spiffworkflow_backend_db:
    driver: local

networks:
  zitiblue:
    external: true
```

**Comment by chrda81** [comment]
I did another try and put the frontend and backend on the same subdomain, but prefix the backend with /path and stript that prefix with traefik on the backend, because it is just virtual. The redirect loop happens 3 or 4 times and the url which is forwarded to the backend is append for this every cycle. This is the resulting string: 

`https://workflow.<domain_name>.com/api/v1.0/login?redirect_url=https://workflow.<domain_name>.com/api/v1.0/login?redirect_url=https%3A%2F%2Fworkflow.<domain_name>.com%2Fapi%2Fv1.0%2Flogin%3Fredirect_url%3Dhttps%253A%252F%252Fworkflow.<domain_name>.com%252Fapi%252Fv1.0%252Flogin%253Fredirect_url%253Dhttps%25253A%25252F%25252Fworkflow.<domain_name>.com%25252Fapi%25252Fv1.0%25252Flogin%25253Fredirect_url%25253Dhttps%2525253A%2525252F%2525252Fworkflow.<domain_name>.com%2525252Fapi%2525252Fv1.0%2525252Flogin%2525253Fredirect_url%2525253Dhttps%252525253A%252525252F%252525252Fworkflow.<domain_name>.com%252525252F`

 Is this something you could explain, if the ping pong is happening in your code between the frontend and backend? Or is it something Traefik or Authentik could do? Here are the essential changes in my docker-compose.yml:

```yml
  spiffworkflow-frontend:
    ...
    environment:
      ...
      SPIFFWORKFLOW_FRONTEND_RUNTIME_CONFIG_BACKEND_BASE_URL: "https://workflow.<domain_name>.com/api"
    ...
    networks:
      - default
      - proxy
    labels:
      traefik.docker.network: proxy
      traefik.enable: true
      traefik.http.routers.spiffWeb-secure.entrypoints: websecure
      traefik.http.routers.spiffWeb-secure.rule: Host(`workflow.starconnect.ch`)
      traefik.http.routers.spiffWeb-secure.service: spiffWeb
      traefik.http.routers.spiffWeb-secure.tls: true
      traefik.http.routers.spiffWeb-secure.tls.certresolver: http_resolver
      traefik.http.routers.spiffWeb.entrypoints: web
      traefik.http.routers.spiffWeb.rule: Host(`workflow.starconnect.ch`)
      traefik.http.services.spiffWeb.loadbalancer.server.port: 8001
      traefik.http.routers.spiffWeb.middlewares: default@file

  spiffworkflow-backend:
    ...
    environment:
      ...
      SPIFFWORKFLOW_BACKEND_URL: "https://workflow.<domain_name>.com/api"
      SPIFFWORKFLOW_BACKEND_URL_FOR_FRONTEND: "https://workflow.<domain_name>.com"
      ...
    networks:
      - default
      - proxy
    labels:
      traefik.docker.network: proxy
      traefik.enable: true
      traefik.http.routers.spiffBackend-secure.entrypoints: websecure
      traefik.http.routers.spiffBackend-secure.rule: Host(`workflow.<domain_name>.com`) && PathPrefix(`/api`)
      traefik.http.routers.spiffBackend-secure.service: spiffBackend
      traefik.http.routers.spiffBackend-secure.tls: true
      traefik.http.routers.spiffBackend-secure.tls.certresolver: http_resolver
      traefik.http.routers.spiffBackend.entrypoints: web
      traefik.http.routers.spiffBackend.rule: Host(`workflow.<domain_name>.com`) && PathPrefix(`/api`)
      traefik.http.services.spiffBackend.loadbalancer.server.port: 8000
      traefik.http.routers.spiffBackend.middlewares: path-stripprefix,default@file
      traefik.http.middlewares.path-stripprefix.stripprefix.prefixes: /api
      ...
```

**Comment by chrda81** [comment]
Thanks to the great support of Kevin, I got it working 🥳

These are the changes which make the connection over https work (otherwise the cookies aren't set on the same domain):

```yml
  spiffworkflow-frontend:
    ...
    environment:
      ...
      # SPIFFWORKFLOW_FRONTEND_RUNTIME_CONFIG_BACKEND_BASE_URL: "https://api.workflow.<domain_name>.com"
    ...
    networks:
      - default
      - proxy
    labels:
      traefik.docker.network: proxy
      traefik.enable: true
      traefik.http.routers.spiffWeb-secure.entrypoints: websecure
      traefik.http.routers.spiffWeb-secure.rule: Host(`workflow.<domain_name>.com`)
      traefik.http.routers.spiffWeb-secure.service: spiffWeb
      traefik.http.routers.spiffWeb-secure.tls: true
      traefik.http.routers.spiffWeb-secure.tls.certresolver: http_resolver
      traefik.http.routers.spiffWeb.entrypoints: web
      traefik.http.routers.spiffWeb.rule: Host(`workflow.<domain_name>.com`)
      traefik.http.services.spiffWeb.loadbalancer.server.port: 8001
      traefik.http.routers.spiffWeb.middlewares: default@file

  spiffworkflow-backend:
    ...
    environment:
      ...
      SPIFFWORKFLOW_BACKEND_URL: "https://api.workflow.<domain_name>.com"
      SPIFFWORKFLOW_BACKEND_URL_FOR_FRONTEND: "https://workflow.<domain_name>.com"
      ...
    networks:
      - default
      - proxy
    labels:
      traefik.docker.network: proxy
      traefik.enable: true
      traefik.http.routers.spiffBackend-secure.entrypoints: websecure
      traefik.http.routers.spiffBackend-secure.rule: Host(`api.workflow.<domain_name>.com`)
      traefik.http.routers.spiffBackend-secure.service: spiffBackend
      traefik.http.routers.spiffBackend-secure.tls: true
      traefik.http.routers.spiffBackend-secure.tls.certresolver: http_resolver
      traefik.http.routers.spiffBackend.entrypoints: web
      traefik.http.routers.spiffBackend.rule: Host(`api.workflow.<domain_name>.com`)
      traefik.http.services.spiffBackend.loadbalancer.server.port: 8000
      traefik.http.routers.spiffBackend.middlewares: default@file
      ...
